getting-started/basicapp/README.md

```diff
@@ -0,0 +1,362 @@
+```
+export MESSAGE_SERVICE=mock
+go get -d -v ./...
+go run basicwebapp.go
+```
+
+Check that the application successfully responds to a HTTP request:
+```
+curl -I http://localhost:8080
+```
+
+You can now use curl to send a message to your friend a message like
+```
+curl "http://localhost:8080/send?user=Friend1&friend=Friend2&text=We+miss+you!"
+```
+
+Your friend can also check their messages with a URL like:
+```
+curl "http://localhost:8080/messages?user=Friend2"
+```
+
+With a mock service we lose the messages as soon as the app is stopped. Unset
+the environment variable for use of mocks with the command
+```
+unset MESSAGE_SERVICE
+```
+
+## Unit tests
+To run unit tests use the two commands, for each of the main and services
+packages:
+```
+go test
+go test -v github.com/GoogleCloudPlatform/golang-samples/getting-started/basicapp/services
+```
+
+Check that the tests pass. The Go
+[httptest](https://golang.org/pkg/net/http/httptest/) package is used to
+simulate HTTP requests and writers in some of the unit tests.
+
+Run the build and execute unit tests using Cloud Build
+```
+gcloud builds submit --config build/cb-unittest.yaml .
+```
+
+View the build and test results in the cloud console under the Cloud Build menu.
+
+## Database
+### Setup
+This section is based on the instructions at [Using Cloud SQL with
+Go](https://cloud.google.com/go/getting-started/using-cloud-sql). Use the
+commands below from the Cloud Shell. Make sure that the Cloud SQL API and the
+Cloud SQL Admin API are enabled.
+
+Create a [Second Generation Cloud SQL 
+instance](https://cloud.google.com/sql/docs/mysql/create-instance) using the 
+commands below, naming the instance 'basicapp'.
+```
+INSTANCE_NAME=basicapp
+MACHINE_TYPE=db-n1-standard-1
+REGION=us-central1
+gcloud sql instances create $INSTANCE_NAME --tier=$MACHINE_TYPE \
+  --region=$REGION
+```
+
+To set the root password, use the cloud console or the command
+```
+gcloud sql users set-password root % --instance basicapp --password [PASSWORD]  
+```
+
+To see the connection details for your instance use the command
+```
+gcloud sql instances describe $INSTANCE_NAME
+```
+
+Look for the ```connectionName```, which as the form 
+```$PROJECT:$REGION:$INSTANCE_NAME```. Also, look for the IP address of the
+instance.
+
+Connect to the your Cloud SQL instance with the command in the cloud shell
+```
+INSTANCE_NAME=basicapp
+gcloud sql connect $INSTANCE_NAME --user=root
+```
+
+Execute the statements in [data/database_setup.sql](data/database_setup.sql).
+
+### Working with the Database in a Local Development Environment (Optional)
+
+To install the Cloud SQL proxy locally follow the steps in [Using Cloud SQL
+with Go](https://cloud.google.com/go/getting-started/using-cloud-sql), for
+example on Mac, use the commands
+
+```
+curl -o cloud_sql_proxy https://dl.google.com/cloudsql/cloud_sql_proxy.darwin.386
+chmod +x cloud_sql_proxy
+```
+
+Start the proxy with the commands
+
+```
+PROJECT=[your project]
```

Comment:
I wonder if this should be `GOOGLE_CLOUD_PROJECT` to be consistent with client libs?

Reply:
Changed to PROJECT_ID to be consistent with Cloud Build and removed GOOGLE_CLOUD_PROJECT elsewhere